modules/openapi-generator/src/main/resources/rust-server/client-mod.mustache

```diff
@@ -246,18 +247,23 @@ impl<F, C> Api<C> for Client<F> where
     C: Has<XSpanIdString> {{#hasAuthMethods}}+ Has<Option<AuthData>>{{/hasAuthMethods}}{
 {{#apiInfo}}{{#apis}}{{#operations}}{{#operation}}
     fn {{#vendorExtensions}}{{{operation_id}}}{{/vendorExtensions}}(&self{{#allParams}}, param_{{{paramName}}}: {{^required}}Option<{{/required}}{{#isListContainer}}&{{/isListContainer}}{{{dataType}}}{{^required}}>{{/required}}{{/allParams}}, context: &C) -> Box<Future<Item={{{operationId}}}Response, Error=ApiError>> {
+        let mut query_string = self::url::form_urlencoded::Serializer::new("".to_owned());
 {{#queryParams}}{{#-first}}
         // Query parameters
-{{/-first}}{{#required}}        let query_{{{paramName}}} = format!("{{{baseName}}}={{=<% %>=}}{<% paramName %>}<%={{ }}=%>&", {{{paramName}}}=param_{{{paramName}}}{{#isListContainer}}.join(","){{/isListContainer}}{{^isListContainer}}.to_string(){{/isListContainer}});
-{{/required}}{{^required}}        let query_{{{paramName}}} = param_{{{paramName}}}.map_or_else(String::new, |query| format!("{{{baseName}}}={{=<% %>=}}{<% paramName %>}<%={{ }}=%>&", {{{paramName}}}=query{{#isListContainer}}.join(","){{/isListContainer}}{{^isListContainer}}.to_string(){{/isListContainer}}));
+{{/-first}}{{#required}}        query_string.append_pair("{{{baseName}}}", &param_{{{paramName}}}{{#isListContainer}}.join(","){{/isListContainer}}{{^isListContainer}}.to_string(){{/isListContainer}});
+{{/required}}{{^required}}        if let Some({{{paramName}}}) = param_{{{paramName}}} {
```

Comment:
This appears to be a change in behaviour. Previously, if we would treat `None` the same as `""`. For example - `foo=bar&baz=&...`.

The new behaviour is to omit optional parameters with a value of `None`. For example - `foo=bar&...`.

Is this intentional? We could probably simplify some of the logic below if we reverted to the old behaviour. I've not used form parameters myself, so am happy to keep the new behaviour if it's more usual.

Reply:
@bjgill From what I've seen optional query parameters are usually handled in a present/not present way instead of empty/non-empty value.

But either way is valid based on [URI spec](https://tools.ietf.org/html/rfc3986#section-3.4), do you think we should change it to empty/non-empty value ?

Reply:
@bjgill I'd vote for removing it. If None is treated as "", there is no semantics to omit the value. If people want to have empty strings they can explicitly provide Some("").

Reply:
An excellent point, @kanekv. The new behaviour does seem most sensible, then.